README.md

```diff
@@ -15,55 +15,110 @@ Clone this repo and `cd` into it.
+
+Rather than allowing all, let's specify which domain is allowed access to a response from our app. Follow these instructions together: 
+
+1. Open the `be/app/controllers/api/v1/welcome_controller.rb` file and look at line 13. This line is commented out. 
+2. Uncomment it, and restart your rails server (`CTRL + C` in Terminal and `rails s` again). Does the request work now? 
+
+This approach _can_ work, but is not a conventional way to solve this problem in Rails. It would also mean we'd have to have this header present on any controller, or any action in each controller, which can quickly get out of hand. 
+
+Let's try a more sustainable approach: 
+
+1. Re-comment out line 13. 
+2. Find `be/config/initializers/cors.rb`. Read this file's instructions and the linked documentation. (By the way, this file was generated by Rails!)
+3. Uncomment lines 8-16, changing `"example.com"` to the origin of your front-end (for our example, that's `"localhost:3000"`), and restart your app. Try the request again. Does the request still work? 
+
+### Questions to answer as a team
+
+Ask these questions out loud, and share answers. Ask your partner to help define any technical terms you are unfamiliar with.
+
+* ❓ What was line 13 in our controller actually doing? 
+* ❓ What is a Header? 
+* ❓ In your own words, what is CORS?
+
+
+## Exploration Activity
 
 In small groups, designate one person to be the screen sharer.
 
-1. FE students, explain to the group what happens when you visit `localhost:3000` in your browser. Start with `index.html` and move to `index.js`. From there, trace all the way through the code up to the point where a request to the BE is made.
-1. BE, explain to the group what happens when the app receives the request. Trace all the way through the code up to the point that a response is sent back to the FE.
-1. FE, explain what happens once the FE receives the response.
-1. Once you are finished exploring the code, draw a diagram that illustrates the entire process.
+1. FE students: explain to the group what happens when you visit `localhost:3000` in your browser. Start with `index.html` and move to `index.js`. From there, trace all the way through the code up to the point where a request to the BE is made.
+1. BE students: explain to the group what happens when your Rails app receives a request. Trace all the way through the code up to the point that a response is sent back to the FE.
+1. FE students:, explain what happens once the FE app receives the response.
+1. Once you are finished exploring the code, draw a diagram that illustrates the entire process. Specify which parts are FE and which are BE. 
+
+
+
+# Extension Ideas
+
+
+### Create & Consume a New API Endpoint
+
+How would you collaborate as a combined team to consume a new set of data? In both applications, implement the ability to list *all* the messages from the BE's database on a new FE page. 
```

Comment:
What about considering this a requirement rather than an extension? I think many people will not think it's necessary when in reality, if they can successfully do this, I think they will be much better off when they start doing Capstone planning.

Reply:
Was thinking similarly - yes, I agree. If we do take the entire afternoon for this project workshop-style (or even async, but for day 1 I kinda feel like we could help with live questions... or even a concurrent office hour), there will probably be time for it. Will make an update..

Reply:
Just updated to include this as a **Practice Activity** at the end!